.github/workflows/release-test-build.yml

```diff
@@ -0,0 +1,177 @@
+            exit 1
+          else
+            echo "Build artifacts present: $(ls -1 dist/ | wc -l) files"
+          fi
+
+      - name: Configure AWS CLI for Cloudflare R2
+        env:
+          R2_ACCOUNT_ID: ${{ secrets.R2_ACCOUNT_ID }}
+          R2_ACCESS_KEY_ID: ${{ secrets.R2_ACCESS_KEY_ID }}
+          R2_SECRET_ACCESS_KEY: ${{ secrets.R2_SECRET_ACCESS_KEY }}
+        run: |
+          aws configure set aws_access_key_id "$R2_ACCESS_KEY_ID"
+          aws configure set aws_secret_access_key "$R2_SECRET_ACCESS_KEY"
+          aws configure set default.region auto
+
+      - name: Upload ${{ matrix.component }} to Cloudflare R2
+        env:
+          R2_ACCOUNT_ID: ${{ secrets.R2_ACCOUNT_ID }}
+          R2_BUCKET_NAME: ${{ secrets.R2_BUCKET_NAME }}
+          RELEASE_ID: ${{ needs.prepare-release.outputs.release_id }}
+        run: |
+          echo "Uploading ${{ matrix.component }} artifacts to R2 bucket: ${R2_BUCKET_NAME}/${RELEASE_ID}/${{ matrix.upload_path }}/"
+          aws s3 sync dist/ "s3://${R2_BUCKET_NAME}/${RELEASE_ID}/${{ matrix.upload_path }}/" \
+            --endpoint-url "https://${R2_ACCOUNT_ID}.r2.cloudflarestorage.com" \
+            --no-progress
+          echo "✓ ${{ matrix.component }} upload complete"
+
+      - name: Clean up workspace
+        if: always()
+        run: rm -rf dist/ || echo "Failed to cleanup workspace"
+
+  notify:
+    needs: [prepare-release, build]
+    runs-on: ubuntu-22.04
+    if: always()
+    steps:
+      - name: Check job status
+        id: check_status
+        run: |
+          if [ "${{ needs.build.result }}" == "success" ]; then
+            echo "status=success" >> $GITHUB_OUTPUT
+          else
+            echo "status=failure" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Configure AWS CLI for Cloudflare R2
+        if: steps.check_status.outputs.status == 'success'
+        env:
+          R2_ACCOUNT_ID: ${{ secrets.R2_ACCOUNT_ID }}
+          R2_ACCESS_KEY_ID: ${{ secrets.R2_ACCESS_KEY_ID }}
+          R2_SECRET_ACCESS_KEY: ${{ secrets.R2_SECRET_ACCESS_KEY }}
+        run: |
+          aws configure set aws_access_key_id "$R2_ACCESS_KEY_ID"
+          aws configure set aws_secret_access_key "$R2_SECRET_ACCESS_KEY"
+          aws configure set default.region auto
+
+      - name: Output summary
+        if: steps.check_status.outputs.status == 'success'
+        env:
+          RELEASE_ID: ${{ needs.prepare-release.outputs.release_id }}
+          R2_BUCKET_NAME: ${{ secrets.R2_BUCKET_NAME }}
+          R2_ACCOUNT_ID: ${{ secrets.R2_ACCOUNT_ID }}
+        run: |
+          R2_URL="https://dash.cloudflare.com/${R2_ACCOUNT_ID}/r2/default/buckets/${R2_BUCKET_NAME}?prefix=${RELEASE_ID}/"
+          echo "## Release Upload Summary" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "**Release ID**: \`${RELEASE_ID}\`" >> $GITHUB_STEP_SUMMARY
+          echo "**Bucket**: \`${R2_BUCKET_NAME}\`" >> $GITHUB_STEP_SUMMARY
```

Comment:
i didnt use GITHUB_STEP_SUMMARY, but should we add R2_BUCKET_NAME to it if its part of secrets?